lib/src/us/kbase/kbasefeaturevalues/KBaseFeatureValuesImpl.java

```diff
@@ -168,10 +177,15 @@ public void clusterKMeans(ClusterKMeansParams params,
         toSave.withFeatureClusters(clustersFromLabels(matrix.getData(), res));
         provenance.get(0).withDescription("K-Means clustering method")
                 .withInputWsObjects(Arrays.asList(params.getInputData()));
-        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(params.getOutWorkspace())
+
+	WorkspaceClient wsc = getWsClient();
+        List<Tuple11<Long,String,String,String,Long,String,Long,String,String,Long,Map<String,String>>> getInfo = wsc.saveObjects(new SaveObjectsParams().withWorkspace(params.getOutWorkspace())
                 .withObjects(Arrays.asList(new ObjectSaveData()
                 .withType("KBaseFeatureValues.FeatureClusters").withName(params.getOutClustersetId())
                 .withData(new UObject(toSave)).withProvenance(provenance))));
+
+        String outref = getInfo.get(0).getE7()  + "/" + getInfo.get(0).getE1() + "/" +  getInfo.get(0).getE5();
```

Comment:
DRY this up, it's repeated multiple times

Reply:
https://github.com/kbaseapps/FeatureValues/pull/21/commits/e6322116fba34b1e3a0060a057ed35f65e9e0e15